recipes-app/iot2050-efivarfs-helper/files/iot2050-efivarfs-helper.tmpl

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) Siemens AG, 2023
+#
+# Authors:
+#  Su Bao Cheng <baocheng.su@siemens.com>
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+import os
+import sys
+import argparse
+import subprocess
+
+
+__version__ = "${PV}"
+
+
+EFIVARFS_MOUNT_POINT = '/sys/firmware/efi/efivars'
+
+
+def err_print(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def exit_on_failure(message, code=1):
+    err_print()
+    err_print(message)
+    err_print()
+    exit(code)
+
+
+def dump_subprocess_error(e: subprocess.CalledProcessError):
+    err_print()
+    err_print("ERROR:\t" + str(e).strip())
+    stdout = str(e.stdout).strip()
+    stderr = str(e.stderr).strip()
+    if stdout != '':
+        err_print("ERROR:\tSTDOUT: " + stdout)
+    if stderr != '':
+        err_print("ERROR:\tSTDERR: " + stderr)
+
+
+def change_efivarfs_immutable_flag(immutable: bool):
+    if immutable:
+        action = 'Adding'
+        option = '+i'
+    else:
+        action = 'Removing'
+        option = '-i'
+
+    for file in os.listdir(EFIVARFS_MOUNT_POINT):
+        if (file.startswith('db-')
+                or file.startswith('KEK-')
+                or file.startswith('PK-')):
+            filepath = os.path.join(EFIVARFS_MOUNT_POINT, file)
+            print("{} the immutable flag of {}... ".format(action, filepath),
+                  end='')
+            cmd = 'chattr {} {}'.format(option, filepath)
+            result = subprocess.run(
+                cmd, shell=True, capture_output=True, check=True, text=True)
+            print("Done.")
+
+
+def do_prepare():
+    message_prepare_fail = "Fail to prepare the efivarfs for manipulating. \
+You may need to check the environment."
+
+    try:
+        mod_required = 'tee_stmm_efi'
+        print("Checking for the kernel module {}... ".format(mod_required),
+              end='')
+        cmd = 'lsmod'
```

Comment:
Hm, can't you just check that `/sys/modules/tee_stmm_efi` is an existing directory? Parsing is always a bit whacky and relies on output format stability.

Reply:
Indeed, checking the file path existence would be a more easy and stable way.